tests/ioctl.c

```diff
@@ -23,25 +23,30 @@ int
 main(void)
 {
 	uint64_t data = 0;
+	uint8_t array[8] = { 0xfe, 0xed, 0xfa, 0xce, 0xde, 0xad, 0xbe, 0xef };
 
 #ifndef POWERPC
 	struct termios tty;
 	(void) ioctl(-1, TCGETS, &tty);
 	printf("ioctl(-1, TCGETS, %p)" RVAL_EBADF, &tty);
 #endif
 
-	(void) ioctl(-1, MMTIMER_GETRES, &data);
-	printf("ioctl(-1, MMTIMER_GETRES, %p)" RVAL_EBADF, &data);
+	(void) ioctl(-1, MMTIMER_GETRES, array);
+	printf("ioctl(-1, MMTIMER_GETRES, "
+	       "\"\\xfe\\xed\\xfa\\xce\\xde\\xad\\xbe\\xef\")"
+	       RVAL_EBADF);
```

Comment:
`MMTIMER_GETRES` is defined to `_IOR(MMTIMER_IOCTL_BASE, 1, unsigned long)` so its _IOC_SIZE is sizeof(long) which is not always 8, that's why the test is failing on 32-bit architectures.

Reply:
Fixed in 56b1d31c51a29ffb261244df6fb216f2e73f89d0.